pkg/lockfile/parse-apk-installed.go

```diff
@@ -0,0 +1,99 @@
+package lockfile
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+)
+
+const AlpineEcosystem Ecosystem = "Alpine"
+
+func groupApkPackageLines(scanner *bufio.Scanner) [][]string {
+	var groups [][]string
+	var group []string
+
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		if line == "" {
+			if len(group) > 0 {
+				groups = append(groups, group)
+			}
+			group = make([]string, 0)
+			continue
+		}
+		group = append(group, line)
+	}
+
+	if len(group) > 0 {
+		groups = append(groups, group)
+	}
+
+	return groups
+}
+
+func parseApkPackageGroup(group []string, pathToLockfile string) PackageDetails {
+	var pkg = PackageDetails{}
+
+	pkg.Ecosystem = AlpineEcosystem
+	pkg.CompareAs = AlpineEcosystem
+
+	// File SPECS: https://wiki.alpinelinux.org/wiki/Apk_spec
+	for _, line := range group {
+		switch {
+		case strings.HasPrefix(line, "P:"):
+			pkg.Name = strings.TrimPrefix(line, "P:")
+		case strings.HasPrefix(line, "V:"):
+			pkg.Version = strings.TrimPrefix(line, "V:")
+		case strings.HasPrefix(line, "c:"):
+			pkg.Commit = strings.TrimPrefix(line, "c:")
+		}
+	}
+
+	if pkg.Version == "" {
+		_, _ = fmt.Fprintf(
+			os.Stderr,
+			"warning: malformed APK installed file. Found no version number in record. File: %s\n",
```

Comment:
nit: I think we should try and include the `name` in here to make it easier to check what package is having problems - we can default so something like `"<unknown>"` if it's blank

Reply:
Hello,
we have two choices, I would go with first one, what you think?

**First one** is `parseApkPackageGroup` continuing to return empty string if package name is empty:
```go
	if pkg.Version == "" {
		var pkgPrintName string
		if pkg.Name == "" {
			pkgPrintName = "<unknown>"
		} else {
			pkgPrintName = pkg.Name
		}

		_, _ = fmt.Fprintf(
			os.Stderr,
			"warning: malformed APK installed file. Found no version number in record for package %s. File: %s\n",
			pkgPrintName,
			pathToLockfile,
		)
	}

	return pkg
}
```
**Second one**, `parseApkPackageGroup` directly returns `<unknown>` but I'm not seeing this string used in other parsers:
```go
	if pkg.Name == "" {
		pkg.Name="<unknown>"
	}
        if pkg.Version == "" {
			_, _ = fmt.Fprintf(
			os.Stderr,
			"warning: malformed APK installed file. Found no version number in record for package %s. File: %s\n",
			pkg.Name,
			pathToLockfile,
		)
	}

	return pkg
}
```

Reply:
yup the first option is the way to go, at least in principle - alternatively you could avoid doing the name check/default by moving the version check to be after the name check [here](https://github.com/google/osv-scanner/pull/107/files#diff-743042e3836e671901997645039e8d1cc69fffb280b9d9377787c072d61c48d9R82).

btw If you decide to stick with your "first one", I would initialize `pkgPrintName` to the value of `pkg.Name` to avoid the `else`

Reply:
Hello @G-Rath ,
followed your advice. Let me know if it's ready for merging.
Thank you again.